### PR TITLE
Add test against the bundled assets

### LINKED ISSUE #208: Test against the bundled assets
At the very beginning i was testing against the build assets, first run build then use the js dist files to run all the tests. After that I switched to [ts alias](https://github.com/huozhi/bunchee/blob/da9df8fb6c376b16197fb6a045b59acace073af1/test/tsconfig.json#L7-L10) that we test against source.

But as a bundler, the output asset could contain problem but the source were good, so I want to have another test that doing the job of using the dist file to test, but will only be able to run after the build.

### PR DESCRIPTION
**Pull request by devjiwonchoi**
Resolves #208

### CODE REVIEW

**Review by huozhi — COMMENTED** [review]
Thanks for the PR!! I think we can able to just switch between `.ts` and `.js` with an process env or sth else so the post-build tests can be only few lines. Sth like

```
process.env.BUNCHEE_DIST = '1'
import '../integration.test'
```

**Comment by devjiwonchoi** [comment]
I like yours much better. Thanks!

File: test/post-build/integration.test.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import '../integration.test'
+
+beforeAll(() => {
+  process.env.POST_BUILD = '1'
```
Comment (huozhi) [inline test/post-build/integration.test.ts:4]: Q: Is `beforeAll` required? not sure what will it effect other tests?
Or do we also need to reset the env afterwards in afterAll?

Comment (devjiwonchoi) [reply test/post-build/integration.test.ts:4]: Seems like beforeAll is not necessary, I'll remove it my bad.

File: test/integration.test.ts [hunk 2]
```diff
@@ -290,10 +290,14 @@ async function runBundle(
   dir: string,
   args_: string[],
 ): Promise<{ code: number | null; stdout: string; stderr: string }> {
+  const assetPath = process.env.POST_BUILD
+    ? '/../dist/cli.js'
+    : '/../src/cli.ts'
+
   const args = (args_ || []).concat(['--cwd', dir])
   const ps = fork(
     `${require.resolve('tsx/cli')}`,
```
Comment (huozhi) [inline test/integration.test.ts:299]: the runner should changed to nodejs instead of tsx/cli

Comment (devjiwonchoi) [reply test/integration.test.ts:299]: Maybe use spawn for js.. or?

```ts
const ps = isPostBuild
  ? spawn('node', [__dirname + assetPath].concat(args), {
      stdio: 'pipe',
    })
  : fork(
      `${require.resolve('tsx/cli')}`,
      [__dirname + assetPath].concat(args),
      {
        stdio: 'pipe',
      },
    )
```

Comment (huozhi) [reply test/integration.test.ts:299]: spawn sounds good 👍

Comment (huozhi) [reply test/integration.test.ts:299]: I realize it would be similar since the input is js so almost no transform, we could change it later

**Review by huozhi — APPROVED** [review]
Good work! Thank you!

**Pull request merged**